### PR TITLE
Refactors TZProfiles to use Rebase SDK

### PR DESCRIPTION
**Pull request by krhoda**
Will need to deploy the Rebase SDK in some public way and update the tzprofiles NPM, then update the package.json to be able to be tested on Cloudflare Build.

### CODE REVIEW

**Comment by krhoda** [comment]
To run locally: (Requires node and typescript compiler)
1. Clone this repo and [rebase](https://github.com/spruceid/rebase) into the same directory. Switch to this branch for this repo and `feat/rebase-sdk-init` for rebase.
2. In rebase run:
```
$ npm i
$ npm run build
```
3. In tzprofiles/contract run:
```
$ tsc
```
4. In tzprofiles/dapp run:
```
$ npm i
$ npm run dev
```

It will be running on `localhost:9080`

File: contract/lib/rebase/tzprofiles.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+export function makeViewClient(opts: ViewClientOpts): ViewClient {
+  return new RebaseViewClient(
+    hasher,
+    makeContentViewer(opts.keplerDomain),
+    makeHashViewer(opts.searchAddress),
+  );
+}
+
+export function viewerOpts(network: NetworkType, ownerAddress: string): ByOwner {
```
Comment (sbihel) [inline contract/lib/rebase/tzprofiles.ts:79]: I'm a bit confused between the viewer that uses the indexer, and this viewer which does the full resolution?

**Review by sbihel — COMMENTED** [review]
I would remove the compiled code (`/dist/`).

There's a lot of code, I'm not sure from which end to start :grimacing:

**Comment by sbihel** [comment]
I'm a bit surprised the "core" rebase code is put in this repo, but globally I don't know how the different parts of Rebase are segregated -- I still need to go over the architecture doc. :bow:

**Pull request closed without merging**